src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -179,4 +165,8 @@ private boolean needToSwitchUser() throws IOException {
     private static boolean isAbsolutePath(String p) {
         return new File(p).isAbsolute();
     }
+
+    private static boolean isQuoted(String s) {
+        return s.startsWith("\"") && s.endsWith("\"") && !s.endsWith("\\\"");
```

Comment:
`|| s.endsWith("\\\\\"")`
Technically I think you need to do some counting to figure out if the final quote is escaped or not.